src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -79,6 +80,34 @@ class PythonSetup(Subsystem):
         advanced=True,
         metavar="<requirement>",
     )
+
+    @memoized_property
```

Comment:
What happens if I have a resolve with an IC - both explicit, but I don't set this? Seems like I both shouldn't have to set it or get nagged for not doing so.

Reply:
I think that would be true if we were sure that any resolve(s) covered all targets, and nothing was relying on this repo-wide default, but I'm not sure how we could know that. Although maybe as a heuristic, if someone has explicitly set an IC for a resolve they probably know what they're doing, so won't need this warning?

Reply:
I also have questions. It seems to me the semantics are confusing and this all just proves that. Perhaps nagging is a step forward, not sure, but my gut says step back and understand what's going on and maybe there is a bigger problem to fix instead of apache fix.

Reply:
We definitely need to look at all this again with the benefit of hindsight. But I think we know with a high degree of confidence that defaulting this value is bad, and so deprecating that is good. It's possible that refactoring or getting rid of it entirely may be even better though.

Reply:
Probably we need to come up with a sensible question and let users answer it, before trying to design anything new here. We are (or at least I am) too poisoned by the status quo.